spec/locales-currencies-tz.html

```diff
@@ -164,59 +164,111 @@ <h1>
     </p>
 
-    <emu-clause id="sec-isvalidtimezonename" type="abstract operation">
-      <h1>
-        IsValidTimeZoneName (
-          _timeZone_: a String,
-        ): a Boolean
-      </h1>
+    <p>This section complements but does not supersede <emu-xref href="#sec-time-zone-identifiers"></emu-xref>.</p>
+
+    <emu-note>
+      <p>
+        The IANA Time Zone Database offers build options that affect which time zone identifiers are primary.
+        The default build options merge different countries' time zones, for example *"Atlantic/Reykjavik"* being a Link to the Zone *"Africa/Abidjan"*.
+        Geographically and politically distinct locations are likely to introduce divergent time zone rules in a future version of the IANA Time Zone Database.
+        Therefore, it is recommended that ECMAScript implementations instead use build options such as <code>PACKRATDATA=backzone PACKRATLIST=zone.tab</code> or a similar alternative that ensures at least one primary identifier for each <a href="https://www.iso.org/glossary-for-iso-3166.html">ISO 3166-1 Alpha-2</a> country code.
+      </p>
+    </emu-note>
+
+    <emu-clause id="sup-availablenamedtimezoneidentifiers" oldids="sec-availabletimezones" type="implementation-defined abstract operation">
+      <h1>AvailableNamedTimeZoneIdentifiers ( ): a List of Time Zone Identifier Records</h1>
       <dl class="header">
         <dt>description</dt>
-        <dd>It verifies that the _timeZone_ argument represents a valid Zone or Link name of the IANA Time Zone Database.</dd>
+        <dd>
+          Its result describes all available named time zone identifiers in this implementation, as well as the primary time zone identifier corresponding to each available named time zone identifier.
+          The List is ordered according to the [[Identifier]] field of each Time Zone Identifier Record.
+        </dd>
+        <dt>redefinition</dt>
+        <dd>true</dd>
       </dl>
+      <p>This definition supersedes the definition provided in <emu-xref href="#sec-availablenamedtimezoneidentifiers"></emu-xref>.</p>
       <emu-alg>
-        1. If one of the Zone or Link names of the IANA Time Zone Database is an ASCII-case-insensitive match for _timeZone_, return *true*.
-        1. If _timeZone_ is an ASCII-case-insensitive match for *"UTC"*, return *true*.
-        1. Return *false*.
+        1. Let _identifiers_ be a List containing the String value of each Zone or Link name in the IANA Time Zone Database.
+        1. Assert: No element of _identifiers_ is an ASCII-case-insensitive match for any other element.
+        1. Assert: Every element of _identifiers_ identifies a Zone or Link name in the IANA Time Zone Database.
+        1. Sort _identifiers_ into the same order as if an Array of the same values had been sorted using %Array.prototype.sort% with *undefined* as the argument.
```

Comment:
```suggestion
        1. Let _identifiers_ be the List of each Zone and Link name in the IANA Time Zone Database, sorted according to lexicographic code unit order.
        1. Assert: No element of _identifiers_ is an ASCII-case-insensitive match for any other element.
        1. Assert: Every element of _identifiers_ identifies a Zone or Link name in the IANA Time Zone Database.
```

I opened https://github.com/tc39/ecma262/pull/3299 to get rid of this awkward way to specify sorting. The language the 262 editors settled on is "sorted according to lexicographic code unit order." That PR has no further comments and I expect it'll be merged this week, so I think it's safe to use that language here as well.